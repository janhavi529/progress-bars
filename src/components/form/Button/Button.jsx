import React from 'react';
import PropTypes from 'prop-types';

const button = ({ className, value, onClick }) => (
    <button type="button" className={className} value={value} onClick={onClick}>{value}</button>
);

button.propTypes = {
    className: PropTypes.string,
    value: PropTypes.number.isRequired,
    onClick: PropTypes.func
};

export default button;