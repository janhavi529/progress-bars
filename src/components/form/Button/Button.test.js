import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    const props = { 
        className: 'btn btn-primary', 
        value: 25, 
        onClick: () => {}
    };

    it('Renders button with default props', () => {
        const wrapper = shallow(<Button {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});