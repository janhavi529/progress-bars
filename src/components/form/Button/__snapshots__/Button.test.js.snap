// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders button with default props 1`] = `ShallowWrapper {}`;
